gensim/models/wrappers/fasttext.py

```diff
@@ -298,8 +299,16 @@ def load_dict(self, file_handle, encoding='utf8'):
                 char_byte = file_handle.read(1)
             word = word_bytes.decode(encoding)
             count, _ = self.struct_unpack(file_handle, '@qb')
-            assert self.wv.vocab[word].index == i, 'mismatch between gensim word index and fastText word index'
-            self.wv.vocab[word].count = count
+
+            if word != "__label__":
+                self.wv.vocab[word] = Vocab(index=i, count=count)
```

Comment:
Is this correct? The word `"__label__"` doesn't have a corresponding vector in the weight matrix, if I understand our previous discussion correctly. As a result, any of the words after the `"__label__"` word will have indices shifted.